pkg/controller/controller.go

```diff
@@ -700,19 +813,37 @@ func (c *controller) instanceDelete(obj interface{}) {
 // Binding handlers and control-loop
 
 func (c *controller) bindingAdd(obj interface{}) {
-	binding, ok := obj.(*v1alpha1.Binding)
-	if binding == nil || !ok {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", obj, err)
 		return
 	}
+	c.bindingQueue.Add(key)
+}
 
-	c.reconcileBinding(binding)
+func (c *controller) reconcileBindingKey(key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
```

Comment:
Likewise, I think we should explain the use of `SplitMetaNamespaceKey` the first place it's used, since it's not at all obvious what `SplitMetaNamespace` means.